### ISSUE TITLE
[New metric] KLDivergence

### ISSUE THREAD

**Issue by SkafteNicki** [PyTorchLightning/metrics#66]
## 🚀 Feature

Add KLDivergence metric. Measures the distance between two probability distributions p(x) and q(x).
Given by (pseudo implementation):
```python
sum(s_p * log(s_p / s_q))
```
where s_p are samples from the p(x) distribution and s_q are samples from the q(x) distribution.
https://en.wikipedia.org/wiki/Kullback%E2%80%93Leibler_divergence

**Comment by hassiahk** [comment]
Hello @SkafteNicki, can I start working on this?

**Comment by SkafteNicki** [comment]
Yes @hassiahk, I will assign you to this issue, thank you for wanting to contribute :]
Please remember to read our contribution guidelines: https://torchmetrics.readthedocs.io/en/stable/generated/CONTRIBUTING.html and this page https://torchmetrics.readthedocs.io/en/stable/pages/implement.html#contributing-your-metric-to-torchmetrics

**Comment by hassiahk** [comment]
Sure, will take a look at them. :) Thanks @SkafteNicki

**Comment by hassiahk** [comment]
Hi @SkafteNicki,

I have seen people using this metric for `regression` tasks but I will be adding this to `classification` because we deal with probabilities and their distributions. Will that be fine?

And how do we test this metric since `sklearn.metrics` does not offer this metric?

**Comment by SkafteNicki** [comment]
Fine by me adding it to `classification`.
You can use http://scipy.github.io/devdocs/generated/scipy.stats.entropy.html function. It calculates the KL-divergence if two distributions are given.

**Comment by SkafteNicki** [comment]
@hassiahk how is it going here? :]

**Comment by hassiahk** [comment]
@SkafteNicki, been busy with some other work, I will create a draft PR for this, this weekend. Thanks!
